Add basic read `consistency` and write `ordering` documentation (#48)
qdrant/docs: qdrant/v1.0.x/distributed_deployment.md

- reads are using a partial fan-out strategy to optimize latency and availability
- writes are executed in parallel on all active sharded replicas

![Embeddings](/docs/concurrent-operations-replicas.png)

However, in some cases, it is necessary to ensure addition guarantees during the possible hardware instability, mass concurrent updates of same documents, etc.

Qdrant provides a few options to control consistency guarantees:

> - `write_concern_factor` - defines the number of replicas that must acknowledge a write operation before responding to the client. Increasing this value will make write operations tollerant to network partitions in the cluster, but will require higher number of replicas to be active to perform write operations.

Reviewer: ```suggestion
- `write_concern_factor` - defines the number of replicas that must acknowledge a write operation before responding to the client. Increasing this value will make write operations tolerant to network partitions in the cluster, but will require a higher number of replicas to be active to perform write operations.
```

> - Read `consistency` param, can be used with search and retrieve operations to ensure that the results obtained from all replicas are the same. In this option is used, qdrant will perform read operation on multiple replicas and resolve the result according to selected strategy. This option is useful to avoid data inconsistency in case of concurrent updates of the same documents. This options is preffered if update operations are frequent and the number of replicas is low.

Reviewer: ```suggestion
- Read `consistency` param, can be used with search and retrieve operations to ensure that the results obtained from all replicas are the same. If this option is used, qdrant will perform the read operation on multiple replicas and resolve the result according to the selected strategy. This option is useful to avoid data inconsistency in case of concurrent updates of the same documents. This options is preferred if the update operations are frequent and the number of replicas is low.
```

> - Write `ordering` param, can be used with update and delete operations to ensure that the operations are executed in the same order on all replicas. If this option is used, qdrant will route the operation to the leader replica of the shard and wait for the response before responding to the client. This option is useful to avoid data inconsistency in case of concurrent updates of the same documents. This options is preffered if read operations are more frequent then update and search performance is critical.

Reviewer: ```suggestion
- Write `ordering` param, can be used with update and delete operations to ensure that the operations are executed in the same order on all replicas. If this option is used, qdrant will route the operation to the leader replica of the shard and wait for the response before responding to the client. This option is useful to avoid data inconsistency in case of concurrent updates of the same documents. This options is preferred if hte read operations are more frequent than update and if search performance is critical.
```

> - Write `ordering` param, can be used with update and delete operations to ensure that the operations are executed in the same order on all replicas. If this option is used, qdrant will route the operation to the leader replica of the shard and wait for the response before responding to the client. This option is useful to avoid data inconsistency in case of concurrent updates of the same documents. This options is preffered if read operations are more frequent then update and search performance is critical.

Reviewer: ```suggestion
- Write `ordering` param, can be used with update and delete operations to ensure that the operations are executed in the same order on all replicas. If this option is used, qdrant will route the operation to the leader replica of the shard and wait for the response before responding to the client. This option is useful to avoid data inconsistency in case of concurrent updates of the same documents. This options is preferred if read operations are more frequent than update and if search performance is critical.
```

)
```

> Write operation will fail if the number of active replicas is less than the `write_concern_factor`.

Reviewer: ```suggestion
Write operations will fail if the number of active replicas is less than the `write_concern_factor`.
```

### Read consistency

Read `consistency` can be specified for most read requests and will ensure that the returned result
is consistent across cluster nodes.

> - `All` will query all nodes and return points, which present on all of them
> - `Majority` will query all nodes and return points, which present on the majority of them
> - `Quorum` will query randomly selected majority of nodes and return points, which present on all of them
> - `N` - will query `N` randomly selected nodes and return points, which present on all of them
> - default `consistency` is `1`

Author: I guess, we should lowercase read-consistency values as well.

```suggestion
- `all` will query all nodes and return points, which present on all of them
- `majority` will query all nodes and return points, which present on the majority of them
- `quorum` will query randomly selected majority of nodes and return points, which present on all of them
- `1`/`2`/`3`/etc - will query specified number of randomly selected nodes and return points which present on all of them
- default `consistency` is `1`
```

```http
POST /collections/{collection_name}/points/search?consistency=majority

{
    "filter": {
        "must": [
            {
                "key": "city",
                "match": {
                    "value": "London"
                }
            }
        ]
    },
    "params": {
        "hnsw_ef": 128,
        "exact": false
    },
    "vector": [0.2, 0.1, 0.9, 0.7],
    "limit": 3
}
```

```python
client.search(
    collection_name="{collection_name}",
    query_filter=models.Filter(
        must=[
            models.FieldCondition(
                key="city",
                match=models.MatchValue(
                    value="London",
                ),
            )
        ]
    ),
    search_params=models.SearchParams(
        hnsw_ef=128,
        exact=False
    ),
    query_vector=[0.2, 0.1, 0.9, 0.7],
    limit=3,
    consistency="majority",
)
```

### Write ordering

Write `ordering` can be specified for any write request to serialize it through a single "leader" node,

> which ensure that all write operations (issued with the same `ordering`) are performed and observed

Reviewer: ```suggestion
which ensures that all write operations (issued with the same `ordering`) are performed and observed
```
